tests/tests_pytorch/utilities/test_data.py

```diff
@@ -328,7 +330,118 @@ def test_replace_dataloader_init_method(cls, args, kwargs, arg_names, dataset, c
         if isinstance(dataloader_value, torch.Tensor):
             assert dataloader_value is value
         else:
-            assert getattr(dataloader, key) == value
+            assert dataloader_value == value
+
+
+def test_replace_init_method_extra_kwargs():
+    class LoaderSubclass(DataLoader):
+        def __init__(self, dataset, *args, batch_size=10, **kwargs):
+            super().__init__(dataset, *args, batch_size=batch_size, **kwargs)
+
+    with _replace_init_method(DataLoader, "dataset"):
+        dataloader = LoaderSubclass(range(10))
+
+    assert dataloader.__pl_saved_args == (range(10),)
+    assert dataloader.__pl_saved_kwargs == {"batch_size": 10}
+    assert dataloader.__pl_saved_arg_names == ("dataset",)
+    assert dataloader.__dataset == range(10)
+
+
+@pytest.mark.parametrize("predicting", [True, False])
+def test_custom_batch_sampler(predicting):
+    class MyBatchSampler(BatchSampler):
+        def __init__(self, sampler, extra_arg, drop_last=True):
+            self.extra_arg = extra_arg
+            super().__init__(sampler, 10, drop_last)
+
+    with _replace_init_method(BatchSampler):
+        sampler = RandomSampler(range(10))
```

Comment:
This instantiation does not need to happen inside the context manager no? Same for the DataLoader.

Same comment applies to the other tests

Reply:
Resolved